### PR TITLE
Remove synthetic flag from extraModifiers

### PR DESCRIPTION
**Pull request by theresa-m**
Since Java 1.5 this flag has been part of modifiers

Fixes: https://github.com/eclipse-openj9/openj9/issues/17785

### CODE REVIEW

File: runtime/bcutil/j9bcu.tdf [hunk 1]
```diff
@@ -76,7 +76,7 @@ TraceException=Trc_BCU_createRomClassEndian_CopyMethodsError Noenv Overhead=1 Le
 TraceException=Trc_BCU_createRomClassEndian_CopyCPShapeError Noenv Overhead=1 Level=3 Template="BCU j9bcutil_createROMClassEndian: error occurred during copyCPShapeAndPreinitDataToROM. error=%d"
 TraceException=Trc_BCU_createRomClassEndian_AnnotationError2 Noenv Overhead=1 Level=3 Template="BCU j9bcutil_createROMClassEndian: error occurred during buildAnnotationInfo. error=%d"
 TraceException=Trc_BCU_createRomClassEndian_DebugInfoError2 Noenv Overhead=1 Level=3 Template="BCU j9bcutil_createROMClassEndian: error occurred during buildDebugInfo. error=%d"
-TraceEvent=Trc_BCU_createRomClassEndian_Synthetic Noenv Overhead=1 Level=3 Template="BCU j9bcutil_createROMClassEndian: Setting J9AccSynthetic flag on ROMClass %p"
+TraceEvent=Trc_BCU_createRomClassEndian_SyntheticOBSOLETE Noenv Overhead=1 Level=3 Template="BCU j9bcutil_createROMClassEndian: Setting J9AccSynthetic flag on ROMClass %p"
```
Comment (pshipton) [inline runtime/bcutil/j9bcu.tdf:79]: `Obsolete` is a keyword, the name of the tracepoint doesn't change. You can search this file for examples.

File: runtime/bcutil/ROMClassBuilder.cpp [hunk 2]
```diff
@@ -1299,12 +1299,6 @@ ROMClassBuilder::computeExtraModifiers(ClassFileOracle *classFileOracle, ROMClas
 		}
 	}
 
-	if ( classFileOracle->isSynthetic() ) {
-		/* handle the synthetic attribute. In java 1.5 synthetic may be specified in the access flags as well so do not unset bit here */
-		// Trc_BCU_createRomClassEndian_Synthetic(romClass);
-		modifiers |= J9AccSynthetic;
```
Comment (hangshao0) [inline runtime/bcutil/ROMClassBuilder.cpp:1305]: Will this cause the below test to fail ?

https://github.com/eclipse-openj9/openj9/blob/7243f7a4e79b039c28fc469fe64557ec46230bad/runtime/bcutil/test/dyntest/romclass_correctness.c#L108-L110

There is another place checking if J9AccSynthetic is set below in `romclass_correctness.c`

Comment (hangshao0) [reply runtime/bcutil/ROMClassBuilder.cpp:1305]: The existing `J9ROMCLASS_IS_SYNTHETIC()` can also be removed.

Comment (theresa-m) [reply runtime/bcutil/ROMClassBuilder.cpp:1305]: I think it should be fine, this looks for the synthetic flag in rom class `modifiers` which isn't effected by this change.

Where does this test code run? I ran a personal build with this change and all sanity.functional tests are passing.

File: runtime/bcutil/j9bcu.tdf (hunk 1 above)
Comment (theresa-m) [reply runtime/bcutil/j9bcu.tdf]: Updated

File: runtime/bcutil/ROMClassBuilder.cpp (hunk 2 above)
Comment (hangshao0) [reply runtime/bcutil/ROMClassBuilder.cpp:1305]: > this looks for the synthetic flag in rom class modifiers which isn't effected by this change

Right, the macro is for `romClass->modifiers`. We should be fine.

**Review by hangshao0 — APPROVED** [review]

**Review by pshipton — APPROVED** [review]

**Pull request merged**